zkllvm-blueprint.nix

```diff
@@ -0,0 +1,128 @@
+    "blueprint_algebra_fields_plonk_non_native_equality_flag_test"
+    "blueprint_algebra_fields_plonk_non_native_division_remainder_test"
+    #blueprint_non_native_plonk_scalar_non_native_range_test, TODO: enable once fixed.
+    "blueprint_non_native_plonk_bool_scalar_multiplication_test"
+    "blueprint_non_native_plonk_add_mul_zkllvm_compatible_test"
+    "blueprint_hashes_plonk_decomposition_test"
+    "blueprint_verifiers_placeholder_fri_cosets_test"
+    "blueprint_hashes_plonk_sha256_process_test"
+    "blueprint_hashes_plonk_sha512_process_test"
+    "blueprint_hashes_plonk_sha256_test"
+    "blueprint_hashes_plonk_sha512_test"
+    "blueprint_algebra_fields_plonk_sqrt_test"
+    "blueprint_verifiers_placeholder_fri_lin_inter_test"
+    "blueprint_verifiers_placeholder_fri_array_swap_test"
+    "blueprint_manifest_test"
+    "blueprint_detail_huang_lu_test"
+    "blueprint_private_input_test"
+    "blueprint_verifiers_placeholder_permutation_argument_verifier_test"
+    "blueprint_verifiers_placeholder_gate_argument_verifier_test"
+    "blueprint_verifiers_placeholder_lookup_argument_verifier_test"
+    "blueprint_verifiers_placeholder_f1_loop_test"
+    "blueprint_verifiers_placeholder_f3_loop_test"
+    "blueprint_verifiers_placeholder_gate_component_test"
+    "blueprint_verifiers_flexible_pow_factor_test"
+    "blueprint_proxy_test"
+    #blueprint_mock_mocked_components_test, TODO: Enable after code and test re-written.
+    "blueprint_component_batch_test"
+    "blueprint_verifiers_placeholder_expression_evaluation_component_test"
+    "blueprint_verifiers_placeholder_final_polynomial_check_test"
+    "blueprint_verifiers_flexible_swap_test"
+    "blueprint_verifiers_flexible_additions_test"
+    "blueprint_verifiers_flexible_multiplications_test"
+    "blueprint_verifiers_flexible_poseidon_test"
+    "blueprint_verifiers_flexible_constant_pow_test"
+    "blueprint_verifiers_placeholder_verifier_test"
+    "blueprint_zkevm_zkevm_word_test"
+    "blueprint_zkevm_bytecode_test"
+    "blueprint_zkevm_state_selector_test"
+    "blueprint_zkevm_state_transition_test"
+    "blueprint_zkevm_opcodes_iszero_test"
+    "blueprint_zkevm_opcodes_add_sub_test"
+    "blueprint_zkevm_opcodes_mul_test"
+    "blueprint_zkevm_opcodes_div_test"
+  ];
+
+  cmakeFlags =
+  [
+      (if runTests then "-DBUILD_TESTS=TRUE" else "")
+      (if runTests then "-DCMAKE_ENABLE_TESTS=TRUE" else "")
+      (if enableDebug then "-DCMAKE_BUILD_TYPE=Debug" else "-DCMAKE_BUILD_TYPE=Release")
+      (if enableDebug then "-DCMAKE_CXX_FLAGS=-ggdb" else "")
+      (if enableDebug then "-DCMAKE_CXX_FLAGS=-O0" else "")
+  ];
+  
+  ninjaFlags = lib.strings.concatStringsSep " " (["-k 0"] ++ testList);
+
+  doCheck = runTests;
+  dontInstall = true;
+
+  checkPhase = ''
+    # JUNIT file without explicit file name is generated after the name of the master test suite inside `CMAKE_CURRENT_SOURCE_DIR` (/build/source)
+    export BOOST_TEST_LOGGER=JUNIT:HRF
```

Comment:
Why do we need to generate junit results at all? Can we drop them?

Reply:
Do you see a pretty report in a comment "Linux Test Results". Those artifacts are required for it.
See step "Publish Test Results" in CI script

Reply:
Why do we need this at all? Shouldn't all tests be green on every pull request?

Reply:
My point is the overcomplication of the whole build and publishing phase is not worth it.